cfroutesync/integration/testenv_test.go

```diff
@@ -275,46 +347,37 @@ users:
 - name: test-user
   user:
     token: %s`, config.Host, config.BearerToken)
-	te.KubeConfigPath = filepath.Join(te.KubeCtlHome, "config")
+	te.KubeConfigPath = filepath.Join(te.TempDir, "kube", "config")
+	if err := os.MkdirAll(filepath.Dir(te.KubeConfigPath), 0644); err != nil {
+		return err
+	}
 	fmt.Fprintf(te.TestOutput, "saving kubecfg to %s\n", te.KubeConfigPath)
 	return ioutil.WriteFile(te.KubeConfigPath, []byte(payload), 0644)
 }
 
-func createCompositeController(webhookHost string) (string, error) {
-	compositeControllerYAML, err := ioutil.TempFile("", "compositecontroller.yaml")
-	if err != nil {
-		return "", err
-	}
-	defer compositeControllerYAML.Close()
-
-	payload := fmt.Sprintf(`---
-apiVersion: metacontroller.k8s.io/v1alpha1
-kind: CompositeController
-metadata:
-  name: cfroutesync
-spec:
-  resyncPeriodSeconds: 5
-  parentResource:
-    apiVersion: apps.cloudfoundry.org/v1alpha1
-    resource: routebulksyncs
-  childResources:
-    - apiVersion: v1
-      resource: services
-      updateStrategy:
-        method: InPlace
-    - apiVersion: networking.istio.io/v1alpha3
-      resource: virtualservices
-      updateStrategy:
-        method: InPlace
-  hooks:
-    sync:
-      webhook:
-        url: http://%s/sync`, webhookHost)
-
-	_, err = compositeControllerYAML.Write([]byte(payload))
+func (te *TestEnv) getResourcesByName(resourceType, namespace string, outMap interface{}) error {
+	out, err := te.kubectl("get", resourceType, "-n", namespace, "-o", "json")
 	if err != nil {
-		return "", nil
+		return err
 	}
+	return k8sListResponseByName(out, outMap)
+}
 
-	return compositeControllerYAML.Name(), nil
+func k8sListResponseByName(rawJSON []byte, outMap interface{}) error {
```

Comment:
This was fascinating, thanks. 🙂

Reply:
Reflection is super powerful.  Generics would be nice though.